pkg/controller/postgresql.go

```diff
@@ -37,6 +51,11 @@ func (c *Controller) clusterListFunc(options api.ListOptions) (runtime.Object, e
 		return nil, fmt.Errorf("could not extract list of postgresql objects: %v", err)
 	}
 
+	if time.Since(c.lastClusterSyncTime) <= c.opConfig.ResyncPeriod {
+		c.logger.Debugln("skipping resync of clusters")
+		return object, err
```

Comment:
I think we should return the list of objects to the shared informer in any case, but only add the `Sync` message not earlier (and, preferably, not later) than after the `ResyncPeriod` from the last sync.